src/main/java/org/opensearch/geospatial/index/query/xyshape/XYShapeQueryProcessor.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.index.query.xyshape;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+
+import org.apache.lucene.document.ShapeField;
+import org.apache.lucene.document.XYShape;
+import org.apache.lucene.geo.XYGeometry;
+import org.apache.lucene.search.MatchNoDocsQuery;
+import org.apache.lucene.search.Query;
+import org.opensearch.common.geo.ShapeRelation;
+import org.opensearch.geometry.Geometry;
+import org.opensearch.geometry.GeometryVisitor;
+import org.opensearch.geospatial.index.mapper.xyshape.XYShapeFieldMapper;
+import org.opensearch.index.mapper.MappedFieldType;
+import org.opensearch.index.query.QueryShardContext;
+import org.opensearch.index.query.QueryShardException;
+
+/**
+ * Query Processor to convert given Geometry into Lucene query
+ */
+public class XYShapeQueryProcessor {
+
+    /**
+     * Creates a {@link Query} that matches all indexed shapes to the provided {@link Geometry}  based on {@link ShapeRelation}
+     * @param geometry OpenSearch {@link Geometry} as an input
+     * @param fieldName Lucene field of type {@link XYShape}
+     * @param relation Relation to be used to get all shapes from given Geometry
+     * @param context QueryShardContext instance
+     * @return {@link Query} instance from XYShape.newGeometryQuery
+     */
+    public Query shapeQuery(
+        Geometry geometry,
+        String fieldName,
+        ShapeRelation relation,
+        GeometryVisitor<List<XYGeometry>, RuntimeException> visitor,
+        QueryShardContext context
+    ) {
+        // validate field type first
+        Objects.requireNonNull(fieldName, "Field name cannot be null");
+        Objects.requireNonNull(context, "QueryShardContext cannot be null");
+        validateIsXYShapeFieldType(fieldName, context);
+
+        // if no input is parsed from input, return no documents should be matched from index,
+        // this is similar to geo_shape field type
+        if (geometry == null || geometry.isEmpty()) {
+            return new MatchNoDocsQuery();
+        }
```

Comment:
This can be moved to beginning to avoid other validations if geometry is null

Reply:
Good point. Currently we validate field name and field type first before checking geometry. If geometry is null, we don’t throw exception, instead, will not return any documents from index. Hence I prioritized validation for exception over geometry. WDYT?

Reply:
Not a big concern. Just that these validations will be unnecessary if ```geometry``` is null. So if possible better to move to beginning.

Reply:
Moved this to beginning of the method.